src/nvim/os/env.c

```diff
@@ -76,6 +76,31 @@ char *os_getenvname_at_index(size_t index)
   return name;
 }
 
+/// Check if the environment variable `name` exists and has
+/// a value of either 1, "on", "yes", or "true".
+bool os_env_istrue(const char *name) FUNC_ATTR_NONNULL_ALL
+{
+  const char *val = os_getenv(name);
+  if (val == NULL || *val == NUL) {  // TODO(Pyrohh): Remove the check for
+    return false;                    // NUL after #2574 is merged.
+  }
+
+  errno = 0;
+  long n = strtol(val, NULL, 10);
+  if (errno == ERANGE || errno == EINVAL) {
+    errno = 0;
+    return false;
+  }
+
+  if (n == 1
+      || strncmp(val, "on", 3) == 0
+      || strncmp(val, "yes", 4) == 0
+      || strncmp(val, "true", 5) == 0) {
+    return true;
```

Comment:
I don't know how accepted this is in C, but I find it very helpful to use the terms "truthy" and "falsey" for  lenient matching like this. E.g. `os_env_istruthy()`.